### PR TITLE
feat: conformance report summary field

### LINKED ISSUE #2235: Conformance Profiles: summaries for profiles without content
**What happened**:

In the last couple of PRs to document some conformance profile reports (#2192 , #2232) I noticed that the "summary" field is often empty for reports.

**What you expected to happen**:

At best, the tooling automates a human readable summary that clearly summarizes the results of the test.

At minimum, let's just not emit empty string fields when the field isn't being set.

### PR DESCRIPTION
**Pull request by mlavacca**
**What type of PR is this?**

/kind feature
/area conformance

**What this PR does / why we need it**:

The conformance report `.profiles[*].summary` field is now automatically filled by the conformance test suite with a human-readable message about the outcome of the tests run.

**Which issue(s) this PR fixes**:

Fixes #2235 

**Does this PR introduce a user-facing change?**:

```release-note
The conformance report `.profiles[*].summary` field is now automatically filled by the conformance test suite with a human-readable message about the outcome of the tests run.
```

### CODE REVIEW

File: conformance/utils/suite/experimental_reports.go [hunk 1]
```diff
@@ -183,3 +185,25 @@ func isTestExtended(profile ConformanceProfile, test ConformanceTest) bool {
...
 }
+
+// buildSummary creates a human-readable message about each profile's test outcomes.
+func buildSummary(report confv1a1.ProfileReport) string {
+	reportMessage := fmt.Sprintf("Core tests %s", buildReportMessage(report.Core))
+	if report.Extended != nil {
+		reportMessage = fmt.Sprintf("%s. Extended tests %s", reportMessage, buildReportMessage(report.Extended.Status))
+	}
+	return fmt.Sprintf("%s.", reportMessage)
+}
```
Comment (shaneutt) [inline conformance/utils/suite/experimental_reports.go:196]: nit, just a suggestion for an alternative:

```suggestion
func buildSummary(report confv1a1.ProfileReport) (reportMessage string) {
	if report.Extended == nil {
		reportMessage = fmt.Sprintf("Core tests %s", buildReportMessage(report.Core))
	} else {
		reportMessage = fmt.Sprintf("%s. Extended tests %s", reportMessage, buildReportMessage(report.Extended.Status))
	}
	return
}
```

File: conformance/utils/suite/experimental_reports.go [hunk 2]
```diff
@@ -183,3 +185,25 @@ func isTestExtended(profile ConformanceProfile, test ConformanceTest) bool {
...
+
+func buildReportMessage(status confv1a1.Status) string {
+	var message string
+	switch status.Result {
+	case confv1a1.Success:
+		message = "succedeed"
```
Comment (shaneutt) [inline conformance/utils/suite/experimental_reports.go:202]:
```suggestion
		message = "succeeded"
```

**Review by shaneutt — COMMENTED** [review]
Cool! Thanks for adding this @mlavacca

File: conformance/utils/suite/experimental_reports.go (hunk 1 above)
Comment (mlavacca) [reply conformance/utils/suite/experimental_reports.go]: It is not the same, as the core results need to be always included in the summary. I've done a small change to the function by following the approach you suggested though: b91df55766cefd366758ca445daa68e187e1209f

**Review by shaneutt — APPROVED** [review]
Thanks @mlavacca!

/approve
/cc @youngnick @robscott

**Comment by mlavacca** [comment]
> Thanks @mlavacca! I assume we've already tested this out on an implementation?

Yes, this has been tested with the Kong Ingress Controller

**Review by sunjayBhatia — APPROVED** [review]
haven't tested it out in a conformance report, but logic and added test cases LGTM

**Comment by robscott** [comment]
Thanks @mlavacca!

/lgtm

**Pull request merged**